frontend/entities/Services.md

```diff
@@ -0,0 +1,21 @@
+# Services
+
+As React is a UI library, we need to ensure to not overload components, providers or any other React entity with too much logic as it can become convoluted and more importantly hard to test.
+
+Services are static modules that provide a list of pure functions that can be used across the application, can be used for calculations or data parsing/modeling.
```

Comment:
I _think_ there's a discussion to be had somewhere about why you're building these as pure functions. Your example below highlights a function with derived data, not really business logic in my opinion. Business logic for this application might be more like deciding whether a user viewing this user's photos should be able to see an individual photo (or any photos at all). For example, if the photo owner uploaded them as private and the person viewing their profile might have certain attributes like not being logged in or not a friend of the photo owner then maybe those photos should be hidden, but if they are logged in and are a friend then the photos might be shown (or not, depending on other unstated criteria, maybe it's not published or it's only visible by the owner). Should that logic live here? Or should it live on the model class? Or should it live on the instance of the model? I know you love JavaScript functional programming but there is a certain sort of classification or grouping that you get from object-oriented programming where certain things that are related live together. I'd be interested to learn more about how you would approach that discussion.

Reply:
It should live in the backend, FE should never take care of things like that

Reply:
But this is a FE Architecture plan 👀